simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -308,11 +319,13 @@ public HTTPServer build() throws IOException {
                 assertNull(sampleNameFilterSupplier, "cannot configure 'sampleNameFilter' and 'sampleNameFilterSupplier' at the same time");
                 sampleNameFilterSupplier = SampleNameFilterSupplier.of(sampleNameFilter);
             }
+
             if (httpServer != null) {
                 assertZero(port, "cannot configure 'httpServer' and 'port' at the same time");
                 assertNull(hostname, "cannot configure 'httpServer' and 'hostname' at the same time");
                 assertNull(inetAddress, "cannot configure 'httpServer' and 'inetAddress' at the same time");
                 assertNull(inetSocketAddress, "cannot configure 'httpServer' and 'inetSocketAddress' at the same time");
+                assertNull(configurator, "cannot configure 'httpServer' and 'configurator' at the same time");
```

Comment:
Nitpick: Replace `'configurator'` with `'httpsConfigurator'` in the error message so that it's clear that this is related to the `withHttpsConfigurator()` call.